datalad/downloaders/tests/test_http.py

```diff
@@ -351,6 +351,13 @@ def request_get_callback(request, uri, headers):
     # the provided URL at the end 404s, or another failure (e.g. interrupted download)
 
 
+@with_testsui(responses=['no', 'yes', 'testlogin', 'testpassword'])
+def test_auth_but_no_cred():
+    authenticator = HTMLFormAuthenticator("")
+    assert_raises(ValueError, HTTPDownloader, credential=None, authenticator=authenticator)
```

Comment:
ok -- create `@with_memory_keyring` decorator, which would mock out `Credential._keyring` to be a `MemoryKeyring()`.  We use that one in `test_credentials` but I never made a decorator for such purpose.  Could be not a decorator but e.g. a context manager so you could do `with memory_keyring as keyring:` and then even interrogate that `keyring` regarding what was stored etc.